src/bootstrap/src/core/build_steps/test.rs

```diff
@@ -783,6 +783,26 @@ impl Step for Clippy {
         let host_libs = builder.stage_out(compiler, Mode::ToolRustc).join(builder.cargo_dir());
         cargo.env("HOST_LIBS", host_libs);
 
+        // Collect paths of tests to run
+        let paths = match &builder.config.cmd {
+            Subcommand::Test { .. } => &builder.config.paths[..],
+            _ => &[],
+        };
```

Comment:
There's no need for pattern matching here. `test::Clippy` is only called with `x test` (i.e., `Subcommand::Test` in the codebase). You can also move this into the `partially_test` block.